File: python/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution:
    def is_prefix(word, prefix):
        if len(word) < len(prefix):
            return False
        for i in range(len(prefix)):
            if word[i] != prefix[i]:
                return False
        return True

    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        i = 1
        for word in sentence.split():
            if Solution.is_prefix(word, searchWord):
                return i
            i += 1
        return -1

File: rust/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.rs
impl Solution {
    fn is_prefix(word: String, prefix: Vec<char>) -> bool {
        if word.len() < prefix.len() {
            return false;
        }
        let word_chars: Vec<char> = word.chars().collect();
        for i in 0..prefix.len() {
            if word_chars[i] != prefix[i] {
                return false;
            }
        }
        return true;
    }

    pub fn is_prefix_of_word(sentence: String, search_word: String) -> i32 {
        let words: Vec<&str> = sentence.split(" ").collect();
        for i in 0..words.len() {
            if Solution::is_prefix(words[i].to_string(), search_word.chars().collect()) {
                return i as i32 + 1;
            }
        }
        return -1;
    }
}
